File: designPatterns/stratagy/StratagyEx2.java
public class StratagyEx2{
  public static void main(String args[]){
    Writer w = new consoleWriter();
    Simulate car = new Simulate(w);
    car.display();
		car.calculate();
  }
}


public class Simulate{
	private Writer writer;
	public Simulate(Writer w){
		this.writer = w;	
	}
  void calculate(){
		writer.write("calculated result");
  }
  public void display(){
		String s = "displaying";
		writer.write(s);
  }
}


public interface Writer{
  public void write(String data);
}
public class consoleWriter implements Writer{
	public void write(String data){
		System.out.println(data);
	}
}

File: designPatterns/stratagy/StratagyEx2.js
class Simulate {
	writer;
	constructor(writer){
		this.writer = writer
	}
	display(){
		const s = "hello"
		this.writer.write(s)
	}
} 
class displayByConsole {
	write(s){
		console.log("displayByConsole "+s)
	}	
}
const sim = new Simulate(new displayByConsole())
sim.display()

// functional way

const simulate = writer => {
	const s = "displaying functional way "
	writer(s)	
}
const writer = s => { console.log(s) }
simulate(writer)

//or 

function Simulate2 (write){
	this.display = () =>{
		write("hello from functinal class")	
	} 
}
const sim2 = new Simulate2(writer)
sim2.display()

